firmware/src/transceiver.c

```diff
@@ -1221,15 +1233,53 @@ void Transceiver_Tasks() {
         PLIB_TMR_Stop(g_hw_settings.timer_module_id);
         ResetToMark();
         g_transceiver.state = STATE_C_COMPLETE;
-      } else {
-        SYS_INT_SourceEnable(g_hw_settings.input_capture_source);
+        return;
       }
+      SYS_INT_SourceEnable(g_hw_settings.input_capture_source);
+      break;
+
+    case STATE_C_RX_IN_MARK:
+      SYS_INT_SourceDisable(g_hw_settings.input_capture_source);
+      if (g_transceiver.state == STATE_C_RX_IN_MARK &&
+          ((uint16_t) (PLIB_TMR_Counter16BitGet(g_hw_settings.timer_module_id) -
+            g_timing.get_set_response.mark_start) >
+            CONTROLLER_RX_MARK_TIME_MAX)) {
+        // Break was too long
+        g_transceiver.result = T_RESULT_RX_INVALID;
+        PLIB_TMR_Stop(g_hw_settings.timer_module_id);
+        ResetToMark();
+        g_transceiver.state = STATE_C_COMPLETE;
+        return;
+      }
+      SYS_INT_SourceEnable(g_hw_settings.input_capture_source);
       break;
 
     case STATE_C_RX_DATA:
-      // TODO(simon): handle the timeout case here.
-      // It's not a static timeout, rather it varies with the slot count.
-      // PLIB_TMR_Stop(g_hw_settings.timer_module_id);
+      // There is not hard timeout on RDM responses, instead it depends on the
+      // number of slots sent, see Table 3.3.
```

Comment:
Okay. The outdated comment was because GitHub hid it when I edited the line above.